### PR TITLE
sysutils/nut: new dashboard widget

### PR DESCRIPTION
**Pull request by DollarSign23**
With the new dashboard, it is now possible to build a widget for NUT. I have based the widget on the design of the new apcupsd widget to keep a consistent look. Thanks to @xbb at this point.

I have seen that a PR for a nut widget has already been created. Just choose the one you like best.

Screenshot from 2024-08-20 07-37-55

### CODE REVIEW

File: sysutils/nut/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PLUGIN_NAME=		nut
-PLUGIN_VERSION=		1.8.1
-PLUGIN_REVISION=	2
+PLUGIN_VERSION=		1.9.0
+PLUGIN_REVISION=	1
```
Comment (fichtner) [inline sysutils/nut/Makefile:3]:
```suggestion
PLUGIN_VERSION=		1.9.0
```

File: sysutils/nut/src/opnsense/www/js/widgets/Metadata/Nut.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <status_fsd>Forced Shutdown</status_fsd>
+            <unconfigured>Nut is not started. Click to configure Nut.</unconfigured>
+            <netclient_unconfigured>This widget only works with the Netclient driver. Click to configure the Netclient driver.</netclient_unconfigured>
+            <netclient_remote_server>Remote NUT Server</netclient_remote_server>
+            <time_hours>h</time_hours>
+            <time_minutes>m</time_minutes>
+            <time_seconds>s</time_seconds>
+        </translations>
+    </nut>
+</metadata>
```
Comment (fichtner) [inline sysutils/nut/src/opnsense/www/js/widgets/Metadata/Nut.xml:42]:
```suggestion
</metadata>

```

File: sysutils/nut/src/opnsense/www/js/widgets/Nut.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    // Translates a key into the corresponding text.
+    translate(key) {
+        let value = this.translations[key];
+        if (value === undefined) {
+            console.error('Missing translation for ' + key);
+            value = key; // Fallback to the key itself if translation is missing.
+        }
+        return value;
+    }
+}
```
Comment (fichtner) [inline sysutils/nut/src/opnsense/www/js/widgets/Nut.js:214]:
```suggestion
}

```

**Comment by fichtner** [comment]
@DollarSign23 nice, thanks! I've asked @mimugmail to review

**Comment by shomilsaini** [comment]
Any updates on when this will be merged to production?

Thanks

File: sysutils/nut/src/opnsense/www/js/widgets/Nut.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (C) 2024 Ben 'DollarSign23'
```
Comment (fichtner) [inline sysutils/nut/src/opnsense/www/js/widgets/Nut.js:2]:
```suggestion
 * Copyright (C) 2024 DollarSign23
```

I'll allow a github handle as copyright, but not a mix of name and handle. Would rather see a full name anyway :)

**Comment by fichtner** [comment]
Merged, thanks!

**Comment by pmhausen** [comment]
Thank you so much, folks. This is incredibly useful!

**Pull request merged**